### ISSUE TITLE
Out of scope datastore displayed incorrectly

### ISSUE THREAD

**Issue by jgadsden** [OWASP/threat-dragon#1101]
**Describe the bug**:
Opening the demo model, the datastore 'Web Application Config' is out of scope, but when first opened it is shown as in scope:

<img width="241" alt="Screenshot 2024-10-06 at 16 35 02" src="https://github.com/user-attachments/assets/8e494768-4577-452b-85f2-0b5879f83800">

only after selecting this component is it properly displayed as out of scope:

<img width="494" alt="Screenshot 2024-10-06 at 16 35 16" src="https://github.com/user-attachments/assets/3339987d-e1f5-47d9-945a-eb65bd3599b4">

**Expected behaviour**:

out of scope elements should always be displayed as such

**Environment**:

- Version: 2.2.0
- Platform: both Web App / Desktop App
- OS: any MacOS / Windows / Linux
- Browser: any chrome, firefox, safari

**To Reproduce**:

open demo model and display the diagram

**Any additional context, screenshots, etc**:

does not seem to apply to dataflows

**Comment by jgadsden** [comment]
tracked down to errors in the ve demo threat model itself

**Comment by Gauravsinghrajputt** [comment]
assign  me for this issues
